File: platforms/kotlin/examples/core/texture_mip_chain/base_size.kt
import org.fragmentcolor.*

val pixels = ByteArray(16 * 16 * 4)
val chain = TextureMipChain.prepare(pixels, TextureFormat.RGBA8_UNORM_SRGB, Size(width=16u, height=16u, depth=null))
val tmp_size = chain.baseSize()
val width = tmp_size.width
val height = tmp_size.height

File: platforms/python/examples/core/texture_mip_chain/base_size.py
from fragmentcolor import TextureFormat, TextureMipChain

pixels = [0] * (16 * 16 * 4)
chain = TextureMipChain.prepare((
    pixels.as_slice(),
    TextureFormat.Rgba8UnormSrgb,
    [16, 16],
))
(width, height) = chain.base_size()
_ = (width, height)